docs(plugins): update plugin documentation, api reference, and developer guide (#542)
falcosecurity/falco-website: content/en/docs/plugins/_index.md

## Plugin Event Sources and Interoperability

> Events returned by source plugins have an "event source" which describes the information in the event. This is distinct from the plugin name to allow for multiple kinds of plugins to generate the same kind of events. For example, there might be plugins gke-audit-bridge, eks-audit-bridge, ibmcloud-audit-bridge, etc. that all fetch [K8s Audit](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/) information. The plugins would have different names and IDs but would have the same event source k8s_audit.

Reviewer: ```suggestion
Events returned by source plugins have an "event source" that describes the event's information. This is distinct from the plugin name to allow for multiple plugin implementations to generate the same kind of events. For example, there might be plugins gke-audit-bridge, eks-audit-bridge, ibmcloud-audit-bridge, etc. that all fetch [K8s Audit](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/) information. The plugins would have different names and IDs but would have the same event source k8s_audit.
```

```yaml
plugins:
  - name: cloudtrail
    library_path: libcloudtrail.so
    init_config: ""
    open_params: ""
  - name: json
    library_path: libjson.so
    init_config: ""

# Optional

> load_plugins: [cloudtrail]

Reviewer: ```suggestion
load_plugins: [cloudtrail, json]
```
Does `cloudtrail` work without `json` ?

Reviewer 2: Well the plugin operates but the rules in aws_cloudtrail_rules.yaml won't. I think we should include both to be a realistic example.

## Plugins

Plugins authored by the Falcosecurity community are at the [plugins github repository](https://github.com/falcosecurity/plugins). The current plugins are:

* **cloudtrail**: Reads Cloudtrail JSON logs from files/S3 and injects as events.
* **dummy**: Reference plugin use to document plugins interface.
* **dummy_c**: Like dummy, but written in C++.

> * **json**: Extracts values from any JSON payload.

Reviewer 2: This is a bit minor, but I actually like listing the plugins here, as it gives a reader a flavor of what's available without having to follow the link. I don't feel super strongly about it, though.

Author: I think we might want to auto-generate this from the plugin registry in the future. Plus, we may also want to re-invent the https://falco.org/docs/event-sources/ section a bit, now that we have plugins. This is more a reminder for ourselves, so for now I'm gonna restore this list!
